src/order/locally_finite.lean

```diff
@@ -646,3 +648,44 @@ begin
 end
 
 end finset
+
+namespace finset
+variables {γ : Type*} [linear_order γ] [locally_finite_order γ]
```

Comment:
I don't even think you need the `locally_finite_order` assumption. Surely any downwards interval of a well order is finite?